Extension/src/LanguageServer/client.ts

```diff
@@ -363,6 +321,62 @@ interface AbortRequestParams {
     id: number;
 }
 
+interface GetSemanticTokensParams {
+    uri: string;
+    id: number;
+}
+
+interface SemanticToken {
+    line: number;
+    character: number;
+    length: number;
+    type: number;
+    modifiers?: number;
+}
+
+interface GetSemanticTokensResult {
+    fileVersion: number;
+    canceled: boolean;
+    tokens: SemanticToken[];
+}
+
+enum SemanticTokenTypes {
+    // These are camelCase as the enum names are used directly as strings in our legend.
+    macro = 0,
+    enumMember = 1,
+    variable = 2,
+    parameter = 3,
+    type = 4,
+    referenceType = 5,
+    valueType = 6,
+    function = 7,
+    member = 8,
+    property = 9,
+    cliProperty = 10,
+    event = 11,
+    genericType = 12,
+    templateFunction = 13,
+    templateType = 14,
+    namespace = 15,
+    label = 16,
+    customLiteral = 17,
+    numberLiteral = 18,
+    stringLiteral = 19,
+    operatorOverload = 20,
+    memberOperatorOverload = 21,
+    newOperator = 22
+}
+
+enum SemanticTokenModifiers {
+    // These are camelCase as the enum names are used directly as strings in our legend.
+    // eslint-disable-next-line no-bitwise
+    static = (1 << 0),
+    // eslint-disable-next-line no-bitwise
+    global = (1 << 1),
+    // eslint-disable-next-line no-bitwise
+    local = (1 << 2)
+}
+
```

Comment:
Interfaces/enums related to colorization or semantic tokens can be placed in separate file and exported to be used in Client class.

Reply:
So, I'm not sold on making these changes.  There is no colorization logic on the TypeScript side anymore, except for logical related to marshaling messages between VS Code and our native process.  All other such marshaling is in this file.  The implementation of the provider is mainly boilerplate (almost identical to code folding provider).  I would agree with a general refactor that moved all providers into separate files, I don't think it makes sense to single out these changes.  I'm open to making the change is others disagree, but would rather do it in a separate PR, and for all providers.

Reply:
Yes, I meant for refactoring the providers into separate files later would be nice.